solidity/dashboard/src/connectors/explorer-mode-connector.js

```diff
@@ -0,0 +1,57 @@
+import { AbstractConnector } from "./abstract-connector"
+import { WALLETS } from "../constants/constants"
+import { getWsUrl } from "./utils"
+import Web3ProviderEngine from "web3-provider-engine"
+import WebsocketSubprovider from "web3-provider-engine/subproviders/websocket"
+import ExplorerModeSubprovider from "./explorerModeSubprovider"
+import { EventEmitter } from "events"
+import { Subprovider } from "@0x/subproviders"
+
+export class ExplorerModeConnector extends AbstractConnector {
+  /** @type {string} To store selected account by user. */
+  selectedAccount = ""
+
+  constructor() {
+    super(WALLETS.EXPLORER_MODE.name)
+    this.explorerModeSubprovider = Object.assign(
+      ExplorerModeSubprovider.prototype,
+      Subprovider.prototype,
+      EventEmitter.prototype
+    )
+  }
+
+  setSelectedAccount = (address) => {
+    this.selectedAccount = address
+    this.emit("accountsChanged", address)
+  }
+
+  enable = async () => {
+    if (!this.selectedAccount) {
+      throw new Error("Account is not selected")
+    }
+
+    this.provider = new Web3ProviderEngine()
+    this.provider.addProvider(this.explorerModeSubprovider)
+    this.provider.addProvider(new WebsocketSubprovider({ rpcUrl: getWsUrl() }))
+
+    this.explorerModeSubprovider.on(
+      "chooseWalletAndSendTransaction",
+      (payload) => {
+        this.emit("chooseWalletAndSendTransaction", payload)
+      }
+    )
+
+    this.provider.start()
+
+    return [this.selectedAccount]
+  }
+
+  disconnect = async () => {
+    this.explorerModeSubprovider.removeListener(
+      "chooseWalletAndSendTransaction",
+      () => {}
```

Comment:
It won't remove listener added in:
```
    this.explorerModeSubprovider.on(
      "chooseWalletAndSendTransaction",
      (payload) => {
        this.emit("chooseWalletAndSendTransaction", payload)
      }
    )
```

It must be the same function, so we need to define this funcion in a class or ` this.explorerModeSubprovider.removeAllListeners("chooseWalletAndSendTransaction")`.

Reply:
Oh, i see! My bad! 😅